qcodes/instrument_drivers/Keysight/N52xx.py

```diff
@@ -0,0 +1,422 @@
+import numpy as np
+from qcodes import VisaInstrument, InstrumentChannel, ArrayParameter, ChannelList
+from qcodes.utils.validators import Numbers, Enum, Bool
+from typing import Sequence, Union, Any, Tuple
+import time
+import re
+
+class PNASweep(ArrayParameter):
+    def __init__(self,
+                 name: str,
+                 instrument: 'PNABase',
+                 **kwargs: Any) -> None:
+
+        super().__init__(name,
+                 instrument=instrument,
+                 shape=(0,),
+                 setpoints=((0,),),
+                 **kwargs
+                 )
+
+    @property # type: ignore
+    def shape(self) -> Sequence[int]: # type: ignore
+        if self._instrument is None:
+            return (0,)
+        return (self._instrument.root_instrument.points(),)
+    @shape.setter
```

Comment:
Do we need these do nothing setters? If we just leave them out setting the value will be an error. Like this it will just do nothing. Is that what we want?

Reply:
If they are not included, then the `super().__init__` call fails. It seemed simpler to create a blank setter than to reimplement the rest of the `__init__` call? Although perhaps it's less clear, and it would be worth reimplementing it anyway...

Reply:
Lets not spent to much time on that. But can you include a comment to mention that